sw.js

```diff
@@ -1,103 +1,85 @@
-  handler: new runtimeCaching.StaleWhileRevalidate({ requestWrapper })
-});
+// Cache fallback for HTML files
+workbox.routing.registerRoute(/(?:^[^.]+|\.html)$/, args =>
+  workbox.strategies
+    .networkFirst()
+    .handle(args)
+    .then(
+      // page is not in cache, respond with 404 page (even if the page exists)
+      response => (response === undefined ? caches.match('/404/') : response)
+    )
+);
 
-/**
- * Route for assets on cloudfront.net CDN
- *
- * Strategy:
- * https://developers.google.com/web/fundamentals/instant-and-offline/
- * offline-cookbook/#stale-while-revalidate
- */
-const cdnAssetRoute = new routing.RegExpRoute({
-  regExp: new RegExp('^https://.*\\.cloudfront\\.net/.*\\.(png|svg)$'),
-  handler: new runtimeCaching.StaleWhileRevalidate({
-    requestWrapper: cdnRequestWrapper
+// Cache CSS and JS files
+workbox.routing.registerRoute(
+  /\.(?:js|css)$/,
+  workbox.strategies.staleWhileRevalidate({
+    cacheName: CACHES.cssJs
   })
-});
-
-/**
- * Route for pages
- *
- * Strategy:
- * https://developers.google.com/web/fundamentals/instant-and-offline/
- * offline-cookbook/#network-falling-back-to-cache
- */
-const navRoute = new routing.NavigationRoute({
-  whitelist: [/./],
-  handler: new runtimeCaching.NetworkFirst({ requestWrapper })
-});
-
-/**
- * Precache resources
- */
-staticCache.addToCacheList({
-  unrevisionedFiles: ['/', '/404/', '/error/']
-});
+);
 
-self.oninstall = event => {
-  event.waitUntil(Promise.all([staticCache.install().then(skipWaiting)]));
-};
+// Cache images
+workbox.routing.registerRoute(
+  /\.(?:png|gif|jpg|jpeg|svg)$/,
+  workbox.strategies.cacheFirst({
+    cacheName: CACHES.images,
+    plugins: [
+      new workbox.expiration.Plugin({
+        maxEntries: 60,
+        maxAgeSeconds: 30 * 24 * 60 * 60 // 30 Days
+      })
+    ]
+  })
+);
 
-self.onactivate = event => {
-  caches.keys().then(keys => {
-    keys.forEach(cacheName => {
-      if (!cacheName.endsWith(version)) {
-        caches.delete(cacheName);
-      }
-    });
-  });
-  return event.waitUntil(clients.claim());
-};
+// Cache Google Fonts
+workbox.routing.registerRoute(
+  new RegExp('https://fonts.(?:googleapis|gstatic).com/(.*)'),
+  workbox.strategies.cacheFirst({
+    cacheName: CACHES.fonts,
+    plugins: [
+      new workbox.expiration.Plugin({
+        maxEntries: 30
+      }),
+      new workbox.cacheableResponse.Plugin({
+        statuses: [200]
+      })
+    ]
+  })
+);
 
-router.registerRoutes({
-  routes: [assetRoute, cdnAssetRoute, navRoute]
-});
+workbox.precaching.precache(['/', '/404/', '/error/']);
 
 /**
- * Fallback route for stuff that doesn't match
+ * Returns true if the cache is one of the specified caches or starts with workbox-
+ * @param {string} cacheName
  */
-router.setDefaultHandler({
-  handler: new runtimeCaching.NetworkOnly()
-});
+const isCacheValid = cacheName =>
+  Object.values(CACHES).includes(cacheName) || cacheName.startsWith('workbox-');
```

Comment:
@calebeby Sorry, I misled you. 🙈 Workbox _does_ handle any caches _it_ creates (the ones that start with `workbox-`). We have to handle any caches we create on our own (e.g. `css-js`, `google-fonts`, `images`).

It should be fine w/o the `|| cacheName.startsWith('workbox-')` here. 😬

Reply:
I wanted to have a whitelist of caches that would be allowed, the rest would be deleted. I allowed the specific ones that we had named, and any starting with `workbox-`. Any other cache was deleted

Reply:
If I didn't whitelist caches starting with `workbox-`, all the workbox caches would be deleted

Reply:
Got it, cool. Thanks for explaining! 👍